dbms/src/WindowFunctions/IWindowFunction.cpp

```diff
@@ -133,6 +133,116 @@ struct WindowFunctionRowNumber final : public IWindowFunction
     }
 };
 
+
+struct WindowFunctionLastValue final : public IWindowFunction
+{
+public:
+    static constexpr auto name = "last_value";
+
+    explicit WindowFunctionLastValue(const DataTypes & argument_types_)
+        : IWindowFunction(argument_types_)
+        , first_unassigned_val_row()
+        , unassigned_row_num(0)
+    {
+        RUNTIME_CHECK(argument_types_.size() == 1);
+        return_type = argument_types_[0];
+    }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    DataTypePtr getReturnType() const override
+    {
+        return return_type;
+    }
+
+    void windowInsertResultInto(
+        WindowTransformAction & action,
+        size_t function_index,
+        const ColumnNumbers & arguments) override
+    {
+        // switch for another frame type
+        switch (action.window_description.frame.end_type)
+        {
+        case WindowFrame::BoundaryType::Current:
+            processFrameTypeCurrent(action, function_index, arguments);
+            break;
+        case WindowFrame::BoundaryType::Unbounded:
+            processFrameTypeUnbounded(action, function_index, arguments);
+            break;
+        case WindowFrame::BoundaryType::Offset:
+            processFrameTypeOffset(action, function_index, arguments);
+            break;
+        default:
+            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Not supported frame BoundaryType");
+        }
+    }
+
+private:
+    static void processFrameTypeCurrent(
+        WindowTransformAction & action,
+        size_t function_index,
+        const ColumnNumbers & arguments)
+    {
+        IColumn & to = *action.blockAt(action.current_row).output_columns[function_index];
+        const auto & value_column = *action.inputAt(action.current_row)[arguments[0]];
+        const auto & value_field = value_column[action.current_row.row];
+        to.insert(value_field);
+    }
+
+    void processFrameTypeUnbounded(
```

Comment:
It seems that all cases can be handled by this function, no need to distinguish which frame type.

Reply:
> It seems that all cases can be handled by this function, no need to distinguish which frame type.

Sry, it's a bad name and the function will be renamed to `processBoundryTypeUnbounded`.